router/src/history/url.rs

```diff
@@ -51,34 +25,31 @@ pub fn escape(s: &str) -> String {
     js_sys::encode_uri(s).as_string().unwrap()
 }
 
-#[cfg(not(feature = "ssr"))]
-impl TryFrom<&str> for Url {
-    type Error = String;
-
-    fn try_from(url: &str) -> Result<Self, Self::Error> {
-        let fake_host = String::from("http://leptos");
-        let url = web_sys::Url::new_with_base(url, &fake_host)
-            .map_err(|e| e.as_string().unwrap_or_default())?;
-        Ok(Self {
-            origin: url.origin(),
-            pathname: url.pathname(),
-            search: url.search(),
-            hash: url.hash(),
-        })
-    }
-}
-
-#[cfg(feature = "ssr")]
 impl TryFrom<&str> for Url {
     type Error = String;
 
     fn try_from(url: &str) -> Result<Self, Self::Error> {
-        let url = url::Url::parse(url).map_err(|e| e.to_string())?;
+        let url = url::Url::parse(&normalize_wasm_url(url)).map_err(|e| e.to_string())?;
         Ok(Self {
             origin: url.origin().unicode_serialization(),
             pathname: url.path().to_string(),
             search: url.query().unwrap_or_default().to_string(),
+            search_params: ParamsMap(
+                url.query_pairs()
+                    .map(|(key, value)| (key.to_string(), value.to_string()))
+                    .collect::<linear_map::LinearMap<String, String>>(),
+            ),
             hash: Default::default(),
         })
     }
 }
+
+#[cfg(feature = "ssr")]
+fn normalize_wasm_url(url: &str) -> Cow<'_, str> {
+    Cow::Borrowed(url)
+}
+
+#[cfg(not(feature = "ssr"))]
+fn normalize_wasm_url(url: &str) -> Cow<'_, str> {
+    Cow::Owned(format!("http://leptos{}", url))
```

Comment:
Kind of hacky. But I am just keeping the current behavior. It seems to me that whoever is trying to convert the URL string to the Url struct should be responsible to guarantee that the string is complete.